spec/models/concerns/searchable_spec.rb

```diff
@@ -1,33 +1,55 @@
 require 'spec_helper'
 
 describe Searchable do
-  describe '#fetch_all' do
-    let(:index_name) { 'test_index_name' }
-    let(:index_type) { 'test_index_type' }
 
-    before do
+    before(:all) do
       class MockModel
-        include Searchable
+        include Indexable
         self.fetch_all_sort_by = 'foo'
-        def self.index_names(_source = nil)
-          ['test_index_name']
+        self.mappings = {
+          name.typeize => {
+            _timestamp: {
+              enabled: true,
+              store:   true,
+            },
+            properties: {
+              foo: { type: 'string' },
+            },
+          }
+        }
+        def self.source
+          {
+            full_name: 'A mocked model',
+            code:      'MModel',
+          }
         end
-        def self.index_types(_source = nil)
-          ['test_index_type']
+        self.model_classes = [MockModel]
+        def self.stored_metadata
+          {version: 123, time: 'a few minutes ago'}
         end
       end
 
-      # I'm doing this without the use of a class which includes Indexable
-      # in order to only use Searchable in this spec.
-      ES.client.indices.delete(index: index_name) if ES.client.indices.exists(index: index_name)
```

Comment:
@calumhalcrow Just want to draw attention to the fact that I've replaced this with actually using Indexable ;-)